cmd/ssv-rewards/calc.go

```diff
@@ -479,12 +522,64 @@ func (c *CalcCmd) populateRewardRedirectsTable(
 	}
 
 	// Verify that the table is populated.
-	count, err = models.RewardRedirects().Count(ctx, c.db)
+	count, err = models.OwnerRedirects().Count(ctx, c.db)
 	if err != nil {
-		return fmt.Errorf("failed to count reward_redirects: %w", err)
+		return fmt.Errorf("failed to count owner_redirects: %w", err)
 	}
 	if int(count) != len(redirects) {
-		return fmt.Errorf("reward_redirects table was not populated")
+		return fmt.Errorf("owner_redirects table was not populated")
+	}
+
+	return nil
+}
+
+func (c *CalcCmd) populateValidatorRedirectsTable(
+	ctx context.Context,
+	redirects rewards.ValidatorRedirects,
+) error {
+	// Truncate the validator_redirects table.
+	_, err := queries.Raw(
+		"TRUNCATE TABLE "+models.TableNames.ValidatorRedirects,
+	).ExecContext(ctx, c.db)
+	if err != nil {
+		return fmt.Errorf("failed to truncate validator_redirects: %w", err)
+	}
+
+	// Verify that the table is empty.
+	count, err := models.ValidatorRedirects().Count(ctx, c.db)
+	if err != nil {
+		return fmt.Errorf("failed to count validator_redirects: %w", err)
+	}
+	if count != 0 {
+		return fmt.Errorf("validator_redirects table was not truncated")
+	}
+
+	// Populate with given redirects.
+	tx, err := c.db.BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to begin transaction: %w", err)
+	}
+	defer tx.Rollback()
```

Comment:
shouldn't we rollback only if tx is failed?

Reply:
> shouldn't we rollback only if tx is failed?

The `defer tx.Rollback()` pattern is directly recommended in the Go database/sql documentation.
[Go database/sql Example](https://pkg.go.dev/database/sql#example-Tx.Prepare)

This ensures that the transaction is safely rolled back in case of errors, and tx.Rollback() after tx.Commit() is a no-op